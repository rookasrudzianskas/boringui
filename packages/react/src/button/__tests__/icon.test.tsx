import React from "react";
import {mount} from "enzyme";

import Button from "../index";

const Icon: React.FC<unknown> = () => <svg />;

describe("ButtonIcon", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Button icon={<Icon />}>action</Button>);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with right", () => {
    const wrapper = mount(<Button iconRight={<Icon />}>action</Button>);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work without text", () => {
    const wrapper = mount(<Button iconRight={<Icon />} />);
    const text = wrapper.find(".nextui-button-text");

    expect(wrapper.html()).toMatchSnapshot();
    expect(text.length).toBe(0);
  });
});
