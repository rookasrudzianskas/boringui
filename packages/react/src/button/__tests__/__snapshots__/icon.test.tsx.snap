// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon should render correctly 1`] = `"<button class=\\"nextui-c-iWjDFM nextui-c-PJLV nextui-c-iWjDFM-hkKLfu-color-default nextui-c-iWjDFM-gkIlwi-size-md nextui-c-iWjDFM-cwXrJp-borderWeight-normal nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-iWjDFM-iPJLV-css nextui-button nextui-button--ready\\" data-state=\\"ready\\" type=\\"button\\"><span class=\\"nextui-c-bDGMeb nextui-c-bDGMeb-iPJLV-css nextui-button-icon\\"><svg></svg></span><div class=\\"nextui-button-text nextui-button-text-left\\">action</div></button>"`;

exports[`ButtonIcon should work with right 1`] = `"<button class=\\"nextui-c-iWjDFM nextui-c-PJLV nextui-c-iWjDFM-hkKLfu-color-default nextui-c-iWjDFM-gkIlwi-size-md nextui-c-iWjDFM-cwXrJp-borderWeight-normal nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-iWjDFM-iPJLV-css nextui-button nextui-button--ready\\" data-state=\\"ready\\" type=\\"button\\"><span class=\\"nextui-c-bDGMeb nextui-c-bDGMeb-jeNDWm-isRight-true nextui-c-bDGMeb-iPJLV-css nextui-button-icon nextui-button-icon-right\\"><svg></svg></span><div class=\\"nextui-button-text nextui-button-text-right\\">action</div></button>"`;

exports[`ButtonIcon should work without text 1`] = `"<button class=\\"nextui-c-iWjDFM nextui-c-PJLV nextui-c-iWjDFM-hkKLfu-color-default nextui-c-iWjDFM-gkIlwi-size-md nextui-c-iWjDFM-cwXrJp-borderWeight-normal nextui-c-iWjDFM-bQrkqa-isChildLess-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-iWjDFM-iPJLV-css nextui-button nextui-button--ready\\" data-state=\\"ready\\" type=\\"button\\"><span class=\\"nextui-c-bDGMeb nextui-c-bDGMeb-jeNDWm-isRight-true nextui-c-bDGMeb-iyYGWO-isSingle-true nextui-c-bDGMeb-bVsTEo-cv nextui-c-bDGMeb-iPJLV-css nextui-button-icon nextui-button-icon-right nextui-button-icon-single\\"><svg></svg></span></button>"`;
